tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -5055,7 +5053,10 @@ class HLSLExternalSource : public ExternalSemaSource {
 
   bool CheckRangedTemplateArgument(SourceLocation diagLoc,
                                    llvm::APSInt &sintValue) {
-    if (!sintValue.isStrictlyPositive() || sintValue.getLimitedValue() > 4) {
+    const auto *SM =
+        hlsl::ShaderModel::GetByName(m_sema->getLangOpts().HLSLProfile.c_str());
+    if (!sintValue.isStrictlyPositive() ||
+        (sintValue.getLimitedValue() > 4 && !SM->IsSM69Plus())) {
```

Comment:
I thought we were not requiring SM 6.9 for long vectors. Am I misreading this or misremembering?

Reply:
Okay, that seems like a reasonable rationale. Separately maybe we should consider if supporting shorter long vectors in old shader models is reasonable. I was looking at some compute shaders last week that bent over backwards to create 8-element vector types which might benefit from a smaller relaxing of vector size limitations.